R/lazyframe-frame.R

```diff
@@ -1841,7 +1842,11 @@ lazyframe__group_by_dynamic <- function(
     every <- parse_as_duration_string(every)
     offset <- parse_as_duration_string(offset) %||% "0ns"
     period <- parse_as_duration_string(period) %||% every
-    group_by <- parse_into_list_of_expressions(!!!group_by)
+    if (!is_polars_expr(group_by)) {
+      group_by <- parse_into_list_of_expressions(!!!group_by)
+    } else {
+      group_by <- list(group_by$`_rexpr`)
+    }
```

Comment:
`parse_into_list_of_expressions()` fails when `group_by` is an expression. It doesn't look great but I can't remember if we had cases like this already

Reply:
An oversight in my review in #49: there are problems with the implementation of `lazyframe__rolling` and `lazyframe__group_by_dynamic` and we need to use dynamic-dots instead of the `group_by` argument.

Reply:
While it's ok to use `...` in `group_by()` because there is no other column selection, the `group_by` argument in `rolling()` and `group_by_dynamic()` is "just" an extra argument and isn't central to using those functions. Therefore, capturing all dots and assign them to `group_by` isn't ideal IMO. It also means that we wouldn't error on misnamed arguments.

Wouldn't it better to accepts expressions in `parse_into_list_of_expressions()` instead, as in py-polars?

Either way, `rolling()` needs to be adjusted.

Reply:
> Wouldn't it better to accepts expressions in `parse_into_list_of_expressions()` instead, as in py-polars?

Since this function here is intended to be passed directly to `...` for ease of use, I think it needs to be processed before passing it to this function rather than having an exception inside this function.

Reply:
Oh, sorry. I think I misunderstood the role of these arguments.
I think this fix should be fine, but I'll look at it some more later.